reconcile/utils/acs_api.py

```diff
@@ -0,0 +1,328 @@
+from typing import (
+    Any,
+    Optional,
+    Self,
+)
+
+import requests
+from pydantic import BaseModel
+
+
+class Role(BaseModel):
+    name: str
+    permission_set_id: str
+    access_scope_id: str
+    description: str
+    system_default: bool
+
+    def __init__(self, api_data: Any) -> None:
+        # attributes defined within stackrox(ACS) API for GET /v1/roles
+        check_len_attributes(
+            ["name", "permissionSetId", "accessScopeId"],
+            api_data,
+        )
+
+        # traits is populated for system default resources and contains `origin: DEFAULT`
+        # this attribute is used to ignore such resources from reconciliation
+        traits = api_data.get("traits")
+        is_default = traits is not None and traits.get("origin") == "DEFAULT"
+
+        super().__init__(
+            name=api_data["name"],
+            permission_set_id=api_data["permissionSetId"],
+            access_scope_id=api_data["accessScopeId"],
+            description=api_data.get("description", ""),
+            system_default=is_default,
+        )
+
+
+class Group(BaseModel):
+    id: str
+    role_name: str
+    auth_provider_id: str
+    key: str
+    value: str
+
+    def __init__(self, api_data: Any) -> None:
+        # attributes defined within stackrox(ACS) API for GET /v1/groups
+        check_len_attributes(["roleName", "props"], api_data)
+        try:
+            check_len_attributes(
+                ["id", "authProviderId", "key", "value"], api_data["props"]
+            )
+        except ValueError as e:
+            # it is valid for the default None group to contain empty key/value
+            if api_data["roleName"] != "None":
+                raise e
```

Comment:
Instead of using error handling to change control flow, how about skip check explicitly?

```python
        if api_data["roleName"] != "None":
            check_len_attributes(
                ["id", "authProviderId", "key", "value"], api_data["props"]
            )
```

Reply:
[updated](https://github.com/app-sre/qontract-reconcile/pull/3899/commits/4546a1fd9cb8560d436edea945dc40fe4207b584)